backend/router/favoritesRouter/favoritesRouter.js

```diff
@@ -3,26 +3,60 @@ import { pool } from '../../helpers/db.js';
 
 const router = Router();
 
-router.post ('/add', async (req, res) => {
+router.post('/add', async (req, res) => {
     try {
-
         const { user_id, fave_1, fave_2, fave_3, fave_4 } = req.body;
- 
-
-pool.query("INSERT INTO favorites (user_id, fave_1, fave_2, fave_3, fave_4)  VALUES ($1, $2, $3, $4, $5)",
-    [user_id, fave_1, fave_2, fave_3, fave_4],
-    (error, results) => {
 
-    if (error) {
+        pool.query("INSERT INTO favorites (user_id, fave_1, fave_2, fave_3, fave_4)  VALUES ($1, $2, $3, $4, $5)",
+            [user_id, fave_1, fave_2, fave_3, fave_4],
+            (error, results) => {
+                if (error) {
+                    console.error("Database query failed:", error.message);
+                    return res.status(500).json({
+                        error: "Database query failed",
+                        details: error.message,
+                    });
+                }
+            }
+        );
+    } catch (error) {
         console.error("Database query failed:", error.message);
         return res.status(500).json({
             error: "Database query failed",
             details: error.message,
         });
     }
 });
-    
 
+// Update favorite
+
+router.put('/update/:id', async (req, res) => {
+    try {
+        const { id } = req.params;
```

Comment:
user_id should be the target, not id

Reply:
Initially, I tried using user_id, but when I entered two records with the same user_id and attempted to update or delete using user_id, it removed both records. That’s why I decided to use id instead. I'm not sure why it allowed the creation of multiple users with the same user_id.

Reply:
you need to make the checks on  /add to make sure multiple rows with the same user_id cannot be created, only 1 row with user_id should ever exist.

update cannot be Id, it has to be user_id because that is what identifies the user to their data